plugins/woocommerce/src/StoreApi/Utilities/OrderController.php

```diff
@@ -168,7 +168,7 @@ public function sync_customer_data_with_order( \WC_Order $order ) {
 	 */
 	public function validate_order_before_payment( \WC_Order $order ) {
 		$needs_shipping          = wc()->cart->needs_shipping();
-		$chosen_shipping_methods = wc()->session->get( 'chosen_shipping_methods' );
+		$chosen_shipping_methods = wc()->session->get( 'chosen_shipping_methods', [ '' ] );
```

Comment:
I think it's best to have this as an empty array and strengthen the checks in `validate_selected_shipping_methods` by separating the check for need_shipping out, and doing array validation second.

Reply:
Done!